src/jit/compiler.h

```diff
@@ -2555,6 +2555,12 @@ class Compiler
     // Returns true iff the secondNode can be swapped with firstNode.
     bool gtCanSwapOrder(GenTree* firstNode, GenTree* secondNode);
 
+    // Given an address expression, compute its costs and addressing mode opportunities,
+    // and mark addressing mode candidates as GTF_DONT_CSE.
+    // TODO-Throughput - Consider actually instantiating these early, to avoid
+    // having to re-run the algorithm that looks for them (might also improve CQ).
+    bool gtMarkAddrMode(GenTree* addr, int& costEx, int& costSz, var_types type);
```

Comment:
I'm not sure if there's a JIT guideline about this but I've seen quite a few C++ devs avoiding such non-const reference parameters and instead using pointers. Pointers have the advantage that they tend to make it more visible at the callsite that the variables may be modified by the call.

Reply:
It is described here https://github.com/dotnet/coreclr/blob/master/Documentation/coding-guidelines/clr-jit-coding-conventions.md#1535-reference-arguments